src/config/backend.ts

```diff
@@ -424,3 +439,218 @@ export class ExtlinuxConfigBackend extends DeviceConfigBackend {
 		return ret;
 	}
 }
+
+export type ConfigfsConfig = Dictionary<string[]>;
+
+/**
+ * A backend to handle ConfigFS host configuration for ACPI SSDT loading
+ *
+ * Supports:
+ * 	- {BALENA|RESIN}_HOST_CONFIGFS_ssdt = value | "value" | "value1","value2"
+ */
+export class ConfigfsConfigBackend extends DeviceConfigBackend {
+	private readonly SystemAmlFiles = path.join(
+		constants.rootMountPoint,
+		'/boot/acpi-tables',
+	);
+	private readonly ConfigFilePath = path.join(bootMountPoint, '/configfs.json'); // use constant for mount path, rename to ssdt.txt
+	private readonly ConfigfsMountPoint = path.join(
+		constants.rootMountPoint,
+		'/sys/kernel/config',
+	);
+	private readonly ConfigVarNamePrefix = `${constants.hostConfigVarPrefix}CONFIGFS_`;
+
+	// supported backend for the following device types...
+	public static readonly SupportedDeviceTypes = ['up-board'];
+
+	private stripPrefix(name: string): string {
+		return name.substr(this.ConfigVarNamePrefix.length);
```

Comment:
I wouldn't want to throw an error, but the usage of this method is either returned or directly compared... so I wouldn't expect any negative fallout from keeping it like this.

```suggestion
		if (!name.startsWith(this.ConfigVarNamePrefix)) {
			return name;
		}
		return name.substr(this.ConfigVarNamePrefix.length);
```

This would make sure the value is only stripped if it's desired though, so I think this is a good option.